metaflow/plugins/argo/argo_workflows.py

```diff
@@ -175,6 +180,44 @@ def deploy(self):
         except Exception as e:
             raise ArgoWorkflowsException(str(e))
 
+    def cleanup_previous(self):
+        try:
+            client = ArgoClient(namespace=KUBERNETES_NAMESPACE)
+            # Check for existing deployment and do cleanup
+            old_template = client.get_workflow_template(self.name)
+            if not old_template:
+                return None
+            # Clean up old sensors
+            old_has_sensor = old_template["metadata"]["annotations"].get(
+                "metaflow/has_sensor"
+            )
+            if old_has_sensor is None:
+                # This workflow was created before sensor annotations
+                # and may have a sensor in the default namespace
+                # we will delete it and it'll get recreated if need be
+                client.delete_sensor(self.name.replace(".", "-"), client._namespace)
+            elif old_has_sensor == "True":
+                # delete old sensor only if it was somewhere else, otherwise it'll get replaced
+                old_sensor_name = old_template["metadata"]["annotations"][
+                    "metaflow/sensor_name"
+                ]
+                old_sensor_namespace = old_template["metadata"]["annotations"][
+                    "metaflow/sensor_namespace"
+                ]
+                if (
+                    not self._sensor
+                    or old_sensor_namespace != ARGO_EVENTS_SENSOR_NAMESPACE
+                ):
+                    client.delete_sensor(old_sensor_name, old_sensor_namespace)
+            # Clean up old schedules/cronworkflow
+            old_has_schedule = old_template["metadata"]["annotations"].get(
+                "metaflow/has_schedule"
+            )
+            if not self._schedule and (old_has_schedule is None or old_has_schedule):
```

Comment:
Only an observation: Previous implementation would simply set the schedule to None I believe, leaving a cron_workflow dangling. This should be a welcome improvement that cleans up the schedule correctly.

Reply:
I reverted this